lib/model/content.dart

```diff
@@ -872,6 +870,53 @@ class _ZulipContentParser {
     return descendant4.text.trim();
   }
 
+  UserMentionNode? parseUserMention(dom.Element element) {
+    assert(_debugParserContext == _ParserContext.inline);
+    assert(element.localName == 'span');
+    final debugHtmlNode = kDebugMode ? element : null;
+
+    final classes = element.className.split(' ')..sort();
+    assert(classes.contains('user-mention')
+        || classes.contains('user-group-mention'));
+    int i = 0;
+
+    if (i >= classes.length) return null;
+    bool isChannelWildcardClassIncluded = false;
+    if (classes[i] == 'channel-wildcard-mention') {
+      // A newly-added class for channel wildcard mentions.
+      // See: https://github.com/zulip/zulip/pull/31075
```

Comment:
I think a comment is useful here, because there is certain information it can convey that's relevant for the reader trying to understand this function. Here's a version focused on that information:
```suggestion
    if (classes[i] == 'channel-wildcard-mention') {
      // Newer channel wildcard mentions have this class; older ones don't.
```

In particular, in my other review comment above I almost suggested naming the variable `isChannelWildcard` — before I remembered that that would be an inaccurate name, because some genuine channel wildcards don't have this class. That fact will be highly relevant when someone goes to implement #646, as it'd otherwise be tempting to use this class for doing so.